src/duty/state/state.metrics.ts

```diff
@@ -0,0 +1,179 @@
+import { Owner, PrometheusService, PrometheusValStatus, TrackTask } from 'common/prometheus';
+import { RegistryService, RegistrySourceOperator } from 'common/validators-registry';
+import { LidoSourceService } from 'common/validators-registry/lido-source';
+import { ClickhouseService } from 'storage/clickhouse';
+
+const GWEI_WEI_RATIO = 1e9;
+
+@Injectable()
+export class StateMetrics {
+  protected epoch: bigint;
+  protected operators: RegistrySourceOperator[];
+  public constructor(
+    @Inject(LOGGER_PROVIDER) protected readonly logger: LoggerService,
+    protected readonly config: ConfigService,
+    protected readonly prometheus: PrometheusService,
+    protected readonly registryService: RegistryService,
+    protected readonly storage: ClickhouseService,
+  ) {}
+
+  @TrackTask('calc-state-metrics')
+  public async calculate(epoch: bigint) {
+    this.logger.log('Calculating state metrics');
+    this.epoch = epoch;
+    this.operators = await this.registryService.getOperators();
+    await Promise.all([
+      this.nosStats(),
+      this.userValidatorsStats(),
+      this.otherValidatorsStats(),
+      this.deltas(),
+      this.minDeltas(),
+      this.negativeValidatorsCount(),
+      this.totalBalance24hDifference(),
+      this.operatorBalance24hDifference(),
+      this.contract(),
+    ]);
+  }
+
+  private async nosStats() {
+    const result = await this.storage.getUserNodeOperatorsStats(this.epoch);
+    for (const nosStat of result) {
+      this.prometheus.userValidators.set(
+        {
+          nos_name: nosStat.val_nos_name,
+          status: PrometheusValStatus.Slashed,
+        },
+        nosStat.slashed,
+      );
+      this.prometheus.userValidators.set(
+        {
+          nos_name: nosStat.val_nos_name,
+          status: PrometheusValStatus.Ongoing,
+        },
+        nosStat.active_ongoing,
+      );
+      this.prometheus.userValidators.set(
+        {
+          nos_name: nosStat.val_nos_name,
+          status: PrometheusValStatus.Pending,
+        },
+        nosStat.pending,
+      );
+    }
+  }
+
+  private async userValidatorsStats() {
+    const result = await this.storage.getUserValidatorsSummaryStats(this.epoch);
+    this.logger.debug(`User ongoing validators [${result.active_ongoing}]`);
+    this.prometheus.validators.set(
+      {
+        owner: Owner.USER,
+        status: PrometheusValStatus.Slashed,
+      },
+      result.slashed,
+    );
+    this.prometheus.validators.set(
+      {
+        owner: Owner.USER,
+        status: PrometheusValStatus.Ongoing,
+      },
+      result.active_ongoing,
+    );
+    this.prometheus.validators.set(
+      {
+        owner: Owner.USER,
+        status: PrometheusValStatus.Pending,
+      },
+      result.pending,
+    );
+  }
+
+  private async otherValidatorsStats() {
+    const result = await this.storage.getOtherValidatorsSummaryStats(this.epoch);
+    this.logger.debug(`Other ongoing validators [${result.active_ongoing}]`);
```

Comment:
Same as above.

Reply:
Useful during debugging and redundant for info log